### ISSUE TITLE
Add convenience `get_sentence_vector()`-like methods for FastText, other models

### ISSUE THREAD

**Issue by gojomo** [RaRe-Technologies/gensim#3015]
Per <https://stackoverflow.com/questions/65397810/whats-the-equivalent-to-get-sentence-vector-for-gensims-fasttext>, the official Python FastText wrapper offers a `get_sentence_vector()` convenience method to averaging the word-vectors of a text. 

Gensim could offer something similar, for FastText & other models. Though, it should perhaps have a more generic name & clear docs that this is just one simple way to create a text-vector from a bunch of words.

**Comment by Rlopezra** [comment]
I like to take a shot on this issue.

**Comment by piskvorky** [comment]
@Rlopezra go for it. Please check the https://github.com/RaRe-Technologies/gensim/wiki/Developer-page first, and let us know how your PR evolves.

**Comment by gojomo** [comment]
@Rlopezra If you are note a regular user of Gensim's word-vector models, please review other relevant issues/discussion-threads/questions to get an idea of what users want this for.

**Comment by Rlopezra** [comment]
@gojomo I'll have an idea of what's being asked but I'll read on  relevant issues/discussion-threads/questions to get a better understanding.

**Comment by AdityaSoni19031997** [comment]
Is this still being considered as a wishlist? I can try to spend some time to fix this;
Thanks.

**Comment by gojomo** [comment]
Yes.

**Comment by rock420** [comment]
I can take this issue if no one is working on it.
>  please review other relevant issues/discussion-threads/questions to get an idea of what users want this for.

Some observations why users want this feature (in the context of finding document similarity)  - 
1.  While doc2vec is a good option for this but in the case of small corpus or shorter documents, it becomes hard for doc2vec algorithm to generate some meaningful embedding for docs. Averaging the word2vec of words in the doc outperform in some cases like that.
2.  Availability of various pre-trained word2vec models, makes it easier to get some useful embedding for a doc by just using the avg of word vectors and in some cases outperform a doc2vec model trained on the corpus.

I think, it will be good if gensim can also provide an option for `TF-IDF` weighted avg vector with this function, as it works better by reducing the influence of common words.

**Comment by gojomo** [comment]
I like the ideo of providing all those options, so long as the method names & doc-comments properly set expectations. 

For example, I'd not want to give the impression that a basic average-of-word-vectors, especially from any arbitrary pretrained word-model, is **the** recommended way to get a vector from a sentence or text – as opposed to just something that's simple and works well in some contexts/baselines. (And of course specifically in the case of a FastText `-supervised` model, combining word-vectors that way works especially well because it matches the training classification goals.)

**Comment by rock420** [comment]
Ok

**Comment by rock420** [comment]
> Though, it should perhaps have a more generic name & clear docs that this is just one simple way to create a text-vector from a bunch of words.

In the current implementation of KeyedVectors, it has `get_vector` as a function to get the vector representation of a key. Should we use something similar to it like `get_avg_vector` (implying the fact that we are taking the average (weighted or non-weighted) for a bunch of given keys) as the function name for our implementation of `get_sentence_vector`?

And regarding the input format, the function should take a list of tokens (pre-processed by user) or a raw sentence (user can provide custom preprocess function as arg also, otherwise gensim would apply some generic preprocessing)?

**Comment by piskvorky** [comment]
> the function should take a list of tokens (pre-processed by user) or a raw sentence

The former (sequence of tokens). We're phasing out preprocessing from Gensim completely, so we definitely don't want to introduce any new preprocessing options. Thanks!

**Comment by gojomo** [comment]
I'd propose:

* because the official FastText Python binding have a `get_sentence_vector()` method, we should define one on the specific `FastTextKeyedVectors` class also, which precisely mimics its behavior - and its doc-comment should note it intends to be a workalike for that
* separately, we should have a utility method on `KeyedVectors` for averaging a bunch of word-vectors from a list of keys/indexes, with helpful options for common cases, such as "pre-normalize" or "ignore missing" or others that seem natural. (Maybe, an optional 'weights' array?) This could be named something like `get_mean_vector()` (not using a abbreviation like `avg`) - and the `get_sentence_vector()` above would make use of this more general method.
*  existing methods which perform an average themselves should where practical be updated to use the new utility method instead. for example, `most_similar()` & `rank_by_centrality()` (used by `doesnt_match()`) each already contain their own mean-of-vectors calculations, which idealy would each use a generalized utility method to avoid repetition, increase clarity, & factor-out the reusable part of the calculation.
